test/e2e_node/init_containers.go

```diff
@@ -0,0 +1,284 @@
+		e2epod.WaitForPodRunningInNamespace(context.TODO(), f.ClientSet, podSpec)
+
+		ginkgo.By("parsing results")
+		results := parseOutput(tmpDir)
+
+		// which we then use to make assertions regarding container ordering
+		ginkgo.By("analyzing results")
+		framework.ExpectNoError(results.StartsBefore(init1, init2))
+		framework.ExpectNoError(results.EndsBefore(init1, init2))
+
+		framework.ExpectNoError(results.StartsBefore(init2, init3))
+		framework.ExpectNoError(results.EndsBefore(init2, init3))
+
+		framework.ExpectNoError(results.StartsBefore(init3, regular1))
+		framework.ExpectNoError(results.EndsBefore(init3, regular1))
+	})
+})
+
+type containerOutput struct {
+	line          int
+	timestamp     string
+	containerName string
+	command       string
+}
+type containerOutputList []containerOutput
+
+func (o containerOutputList) String() string {
+	var b bytes.Buffer
+	for _, v := range o {
+		fmt.Fprintf(&b, "%d %s %s %s\n", v.line, v.timestamp, v.containerName, v.command)
+	}
+	return b.String()
+}
+
+// StartsBefore returns an error if lhs did not start before rhs
+func (o containerOutputList) StartsBefore(lhs, rhs containerTestConfig) error {
+	lhsStart := o.findIndex(lhs.Name, "Starting")
+	rhsStart := o.findIndex(rhs.Name, "Starting")
+
+	if lhsStart == -1 {
+		return fmt.Errorf("couldn't find that container %s ever started, got %s", lhs.Name, o)
+	}
+	if rhsStart == -1 {
+		return fmt.Errorf("couldn't find that container %s ever started, got %s", rhs.Name, o)
+	}
+	if lhsStart >= rhsStart {
+		return fmt.Errorf("expected container %s to start before %s, got %s", lhs.Name, rhs.Name, o)
+	}
+	return nil
+}
+
+// EndsBefore returns an error if lhs did not end before rhs
+func (o containerOutputList) EndsBefore(lhs, rhs containerTestConfig) error {
+	lhsExit := o.findIndex(lhs.Name, "Exiting")
+	rhsExit := o.findIndex(rhs.Name, "Exiting")
+
+	if lhsExit == -1 {
+		return fmt.Errorf("couldn't find that container %s ever exited, got %s", lhs.Name, o)
+	}
+	if rhsExit == -1 {
+		return fmt.Errorf("couldn't find that container %s ever exited, got %s", rhs.Name, o)
+	}
+	if lhsExit >= rhsExit {
+		return fmt.Errorf("expected container %s to exit before %s, got %s", lhs.Name, rhs.Name, o)
+	}
+	return nil
+}
+
+func (o containerOutputList) findIndex(name string, command string) int {
+	for _, v := range o {
+		if v.containerName == name && v.command == command {
+			return v.line
+		}
+	}
+	return -1
+}
+
+func parseOutput(dir string) containerOutputList {
+	contents, err := os.ReadFile(filepath.Join(dir, "output"))
+	framework.ExpectNoError(err, "reading output file")
+
+	sc := bufio.NewScanner(bytes.NewReader(contents))
+	var res containerOutputList
+	lineNo := 0
+	for sc.Scan() {
+		lineNo++
+		fields := strings.Fields(sc.Text())
+		if len(fields) < 3 {
+			framework.ExpectNoError(fmt.Errorf("%v should have at least length 3", fields))
+		}
+		res = append(res, containerOutput{
+			line:          lineNo,
+			timestamp:     fields[0],
```

Comment:
Might parse this to an int and use this at some point, busybox doesn't support %N so we only have 1 second resolution here.

Reply:
This does work out to produce something that's nicely testable.